From 899bae762921d08dd4552deb52df84afafe00abb Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Fri, 6 Nov 2015 14:41:33 -0800
Subject: [PATCH 228/441] Revert "usb: dwc3: HACK: assert ulpi_refclk"

This reverts commit 5730a3e303fe2c890b9f836d77cec2f407d13d34.

This is not needed on Cougar Mountain, as the BIOS already enables the
ULPI refclock. Also, BAR1 is not exposed, so this function will fail to
write the register.

HSD: 244432

Signed-off-by: Brett T. Warden <brett.t.warden@intel.com>
---
 drivers/usb/dwc3/dwc3-pci.c | 37 -------------------------------------
 1 file changed, 37 deletions(-)

--- a/drivers/usb/dwc3/dwc3-pci.c
+++ b/drivers/usb/dwc3/dwc3-pci.c
@@ -116,36 +116,6 @@ err1:
 	return ret;
 }
 
-#define GP_RWREG1			0xa0
-#define GP_RWREG1_ULPI_REFCLK_DISABLE	(1 << 17)
-static void dwc3_pci_enable_ulpi_refclock(struct pci_dev *pci)
-{
-	void __iomem	*reg;
-	struct resource	res;
-	struct device	*dev = &pci->dev;
-	u32 		value;
-
-	res.start	= pci_resource_start(pci, 1);
-	res.end 	= pci_resource_end(pci, 1);
-	res.name	= "dwc_usb3_bar1";
-	res.flags	= IORESOURCE_MEM;
-
-	reg = devm_ioremap_resource(dev, &res);
-	if (IS_ERR(reg)) {
-		dev_err(dev, "cannot check GP_RWREG1 to assert ulpi refclock\n");
-		return;
-	}
-
-	value = readl(reg + GP_RWREG1);
-	if (!(value & GP_RWREG1_ULPI_REFCLK_DISABLE))
-		return; /* ULPI refclk already enabled */
-
-	/* Let's clear ULPI refclk disable */
-	dev_warn(dev, "ULPI refclock is disable from the BIOS, let's try to enable it\n");
-	value &= ~GP_RWREG1_ULPI_REFCLK_DISABLE;
-	writel(value, reg + GP_RWREG1);
-}
-
 static int dwc3_pci_probe(struct pci_dev *pci,
 		const struct pci_device_id *id)
 {
@@ -223,13 +193,6 @@ static int dwc3_pci_probe(struct pci_dev
 	dwc3->dev.parent = dev;
 	glue->dwc3 = dwc3;
 
-	/*
-	 * HACK: we found an issue when enabling DWC3 controller where the
-	 * refclock to the phy is not being enabled.
-	 * We need an extra step to make sure such clock is enabled.
-	 */
-	dwc3_pci_enable_ulpi_refclock(pci);
-
 	ret = platform_device_add(dwc3);
 	if (ret) {
 		dev_err(dev, "failed to register dwc3 device\n");
